File: CloudFlow/FrontEnd-App/src/app/shared/model/content/nomenclator-metadata-definition-value-selection-filter.model.ts
import { JsonObject, JsonProperty } from "json2typescript";

@JsonObject
export class NomenclatorMetadataDefinitionValueSelectionFilterModel {
	
	@JsonProperty("id", Number)
	public id: number = null;

	@JsonProperty("filterAttributeId", Number)
	public filterAttributeId: number = null;

	@JsonProperty("filterAttributeKey", String, true)
	public filterAttributeKey: string = null;

	@JsonProperty("defaultFilterValue", String, true)
	public defaultFilterValue: string = null;

	@JsonProperty("metadataNameForAutocompleteFilterValue", String, true)
	public metadataNameForAutocompleteFilterValue: string = null;
	
	public clone(): NomenclatorMetadataDefinitionValueSelectionFilterModel {
		let clone: NomenclatorMetadataDefinitionValueSelectionFilterModel = new NomenclatorMetadataDefinitionValueSelectionFilterModel();
		Object.keys(this).forEach((prop: string) => {
			clone[prop] = this[prop];
		});
		return clone;
	}
}

File: CloudFlow/src/ro/cloudSoft/cloudDoc/presentation/client/shared/model/content/NomenclatorMetadataDefinitionValueSelectionFilterModel.java
package ro.cloudSoft.cloudDoc.presentation.client.shared.model.content;

public class NomenclatorMetadataDefinitionValueSelectionFilterModel {
	
	private Long id;	
	private Long filterAttributeId;
	private String filterAttributeKey;
	private String defaultFilterValue;
	private String metadataNameForAutocompleteFilterValue;
	
	public Long getId() {
		return id;
	}
	public void setId(Long id) {
		this.id = id;
	}
	public Long getFilterAttributeId() {
		return filterAttributeId;
	}
	public void setFilterAttributeId(Long filterAttributeId) {
		this.filterAttributeId = filterAttributeId;
	}
	public String getFilterAttributeKey() {
		return filterAttributeKey;
	}
	public void setFilterAttributeKey(String filterAttributeKey) {
		this.filterAttributeKey = filterAttributeKey;
	}
	public String getDefaultFilterValue() {
		return defaultFilterValue;
	}
	public void setDefaultFilterValue(String defaultFilterValue) {
		this.defaultFilterValue = defaultFilterValue;
	}
	public String getMetadataNameForAutocompleteFilterValue() {
		return metadataNameForAutocompleteFilterValue;
	}
	public void setMetadataNameForAutocompleteFilterValue(String metadataNameForAutocompleteFilterValue) {
		this.metadataNameForAutocompleteFilterValue = metadataNameForAutocompleteFilterValue;
	}
}